### ISSUE TITLE
Typescript import doesn't work

### ISSUE THREAD

**Issue by polRk** [i18next/i18next#1271]
```
import i18next from 'i18next'

import { enLocales } from './locales/en'
import { ruLocales } from './locales/ru'

console.log(i18next) // undefined

i18next.init({
  debug: true,
  resources: { en: enLocales, ru: ruLocales }
})

```

```
i18next_1.default.init({
                  ^

TypeError: Cannot read property 'init' of undefined
```

tsconfig
```

{
  "compilerOptions": {
    "lib": [
      "esnext",
      "dom"
    ],
    "module": "commonjs",
    "noImplicitReturns": true,
    "noUnusedLocals": true,
    "outDir": "lib",
    "sourceMap": true,
    "strict": true,
    "target": "es2017"
  },
  "compileOnSave": true,
  "include": [
    "src"
  ]
}

```
I can not use esModuleInterop, because many other libraries fall back

_Originally posted by @polRk in https://github.com/i18next/i18next/issues/1177#issuecomment-504989651_

**Comment by polRk** [comment]
```
import i18next = require('i18next');

console.log(i18next.default) // undefined
```

**Comment by polRk** [comment]
```
import * as i18next from 'i18next'

console.log(i18next.default) // undefined
```

**Comment by rosskevin** [comment]
I believe this is because the original types imported are hidden inside a namespace. The namespace should be removed. PR is welcome.

**Comment by polRk** [comment]
I think you should correct this mistake

**Comment by rosskevin** [comment]
@polRk I think you should have some consideration for open source maintainers and learn that you need to contribute fixes for your own problems.

I volunteer to help with this project because I use it. It seems you are using it and have a problem that you want fixed, therefore I am willing to guide you in solving it.

TL;DR I am not responsible for fixing problems you have with this library, you are, and I am willing to help you.

**Comment by polRk** [comment]
Why i can import moment, date-fns, react, angular, vue and so on... ? )

**Comment by rosskevin** [comment]
The difference is a stylistic choice from the original types authors. Namespaces still have their use but fairly quickly fell out of favor. It is not "wrong" but different. Fixing this is a matter of removing the namespace, altering tests, and altering any changes in downstream projects like react-i18next.

**Comment by koloboid** [comment]
```
import i18default from "i18next";
const i18next: typeof i18default = require("i18next");
```

Simple workarround for those who cannot use esModuleInterop

**Comment by johannesschobel** [comment]
i am currently struggling with the same issue. The `7.0.1` version works for me, `7.0.2` and upwards does not work any more..
Unfortunately, i cannot use the `esModuleInterop: true` compiler flag in the `tsconfig.json` file, as other packages then fail..

Downside with the approach mentioned above is, that you need to do this every time you access the `i18next` package (i.e., in different files!)

Can we provide a better solution for this issue?

**Comment by rosskevin** [comment]
@johannesschobel - better solution already mentioned above:

> I believe this is because the original types imported are hidden inside a namespace. The namespace should be removed. PR is welcome.

If it is important for you, then it should be important enough for you to make an attempt and submit a PR.  Open source requires effort from all including maintainers and users.

**Comment by johannesschobel** [comment]
@rosskevin , yeah sure - and i am fully on your side with this.. however, i don't feel competent enough to fix this in order to not break something else. therefore, i cannot contribute with a PR - i am very sorry..
if i would be able, i would sure send a PR, because i would like to use the latest version (and the `@next` ones as well)

**Comment by tmtron** [comment]
I recommend to reopen this issue: then you are at least aware of it and maybe some good soul will eventually come up with a PR.  

IMHO it certainly needs to be addressed, because the [workaround ](https://github.com/i18next/i18next/issues/1271#issuecomment-506946852) we need is nasty and unexpected. At least for me: We use lots of packages and I've never had to use `require` before.  

When you don't want to fix it for any reason, 
* you should at least document how this works, for other newbie-typescript uses, like me
* and since you have a [Migration Guide](https://www.i18next.com/misc/migration-guide) this change is certainly be something to mention, because it broke your existing code when upgrading from 15.x to 17.x

**Comment by jamuhl** [comment]
@tmtron both the documentation and the code can be improved by creating a PR. I personally never used typescript. But in the last year, I at least opened the repository for typescript related contributions - but the quality of it lives from you...

**Comment by koloboid** [comment]
@tmtron absolutely agree with you, issue have to be reopen, workaround is actually workaround and temporary solution. `require` is weird in TypeScript, while setting esModuleInterop breaks even hello world projects

**Comment by rosskevin** [comment]
Typescript is maintained by the community of users, not @jamuhl . If you want something, PR it and tag me. It is is really as simple as that.

**Comment by tmtron** [comment]
> ```
> import i18default from "i18next";
> const i18next: typeof i18default = require("i18next");
> ```
> 
> Simple workarround for those who cannot use esModuleInterop

The **`require` [workaround](https://github.com/i18next/i18next/issues/1271#issuecomment-506946852)** only works for me on the server-side, but it **does not work on the client-side** (angular 8 app). I guess this is related to some typescript compiler options: e.g. the target is different: `es5` vs. `es2015`?   
Thus I had to revert i18next back to `15.1.3`

any ideas for a workaround that works for both cases?

**Comment by tmtron** [comment]
I got it working with version `17.0.17` after activating the `esModuleInterop` ts-config option (needed some changes to other imports).
 According to [Typescript 2.7 release notes](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#support-for-import-d-from-cjs-from-commonjs-modules-with---esmoduleinterop), we should anyway always activate `esModuleInterop`.

The import statement is now: `import i18next from 'i18next';`

In our `tsconfig.json` we had `"allowSyntheticDefaultImports": true`, but `esModuleInterop` was not defined (default=`false`): I think this combination caused the issues in our code-base.

**Comment by falahati** [comment]
If you can't enable `esModuleInterop`, you can always try this dirty hack to get the typing to work:
import * as i18next from 'i18next'
const i18Next: i18next.i18n = i18next as any as i18next.i18n;

**Comment by rosskevin** [comment]
Once again, you should not need this with the latest release unless you are loading the commonjs package and refuse to follow typescript's recommendation to enable `esModuleInterop` which is exactly intended for this use case.

See #1352
